proposals/explicit-backing-fields.md

```diff
@@ -14,417 +14,396 @@ but one is part of a public API and another is an implementation detail.
+     | (setter? (semi? getter)? (semi? field)?)
+     | (getter? (semi? field)? (semi? setter)?) 
+     | (setter? (semi? field)? (semi? getter)?)
+     | (field? (semi? getter)? (semi? setter)?) 
+     | (field? (semi? setter)? (semi? getter)?))
 
-> In above example `items` is `MutableList<Item>` when accessed privately inside class and read-only `List<String>` when accessed from outside class.
+field ::=
+    modifiers? 'field' (':' type)? '=' expression     
+```
 
-In fact, the above syntax brings in an incorrect mental model: it says _'There is a private property `items` with some *part* that declares its public behavior'_.
+### Java interop
 
-Attempt to add support for the above syntax led to multiple redundant complications (see the [problems section](https://github.com/matejdro/KEEP/blob/private_public_property_types/proposals/private_public_property_types.md#questions-to-consider) and unclear override mechanics).
+For JVM target, property with explicit backing field is compiled into field(-s) and accessors.
+Property can be accessed from Java through them in accordance with the following visibility rules:
 
-## Future Enhancements
+* accessors are assigned visibility of property
+* field is assigned `private` visibility of explicit backing field 
 
-### Direct Backing Field Access
-                                                                     
-We plan to add support for a syntax to explicitly access the property's backing field as well as for property 
-delegate via some TBD syntax.  
+```kotlin
+protected val name: LiveData<String>
+    field = MutableLiveData<String>()
+```
+
+So, the property above is visible in Java as:
 
-###  Protected Fields
-                     
-The set of visibilities for an explicitly declared field can be extended to include `protected`
-(in addition to `private` and `internal`). This way, subclasses can explicitly or implicitly
-(via the [smart type narrowing](#smart-type-narrowing)) reference the field.
+```java
+private final MutableLiveData<String> name = MutableLiveData();
 
-### Mutable Fields for Read-only Properties
+protected final LiveData<String> getName() {
+    return this.name;
+}
+```
 
-Letting a `val` property have a mutable backing field may be useful. 
-Consider the following snippet from the 
-[implementation of `lazy` in the Kotlin standard library](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/jvm/src/kotlin/util/LazyJVM.kt#L57):
+`@JvmField` is prohibited on properties with explicit backing field, because in this case a field type becomes exposed,
+which makes use of the feature completely pointless.
+
+#### Property with initializer compilation
+
+Property with explicit backing field and initializer is represented by additional
+auxiliary synthetic field which is needed for storing the value from property initializer.
+Auxiliary field is given the name `propertyName$init` and is `private` regardless of property visibility.
```

Comment:
It may be worth mentioning if allowing everything (accessors + access to both the fields as `field` and `init` within them) is considered a future enhancement or something we'd like to avoid for design reasons